examples/embassy-http-server/src/main.rs

```diff
@@ -43,7 +38,25 @@ impl picoserve::extract::FromRef<AppState> for ButtonInputs {
     }
 }
 
-type AppRouter = impl picoserve::routing::PathRouter<AppState>;
+mod approuter {
```

Comment:
This refactor is needed for the latest TAIT changes to prevent this:

<details>

```
error: item does not constrain `AppRouter::{opaque#0}`, but has it in its signature                                                                            
  --> examples/embassy-http-server/src/main.rs:51:10                                                                                                           
   |                                                                                                                                                           
51 | async fn web_task(                                                                                                                                        
   |          ^^^^^^^^                                                                                                                                         
   |                                                                                                                                                           
   = note: consider moving the opaque type's declaration and defining uses into a separate module                                                              
note: this opaque type is in the signature                                                                                                                     
  --> examples/embassy-http-server/src/main.rs:46:18                                                                                                           
   |                                                                                                                                                           
46 | type AppRouter = impl picoserve::routing::PathRouter<AppState>;                                                                                           
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                            
                                                                                                                                                               
error: item does not constrain `AppRouter::{opaque#0}`, but has it in its signature                                                                            
  --> examples/embassy-http-server/src/main.rs:56:8                                                                                                            
   |                                                                                                                                                           
56 |   ) -> ! {                                                                                                                                                
   |  ________^
57 | |     let stack = network::network_stack().await.unwrap();
58 | |
59 | |     let mut rx_buffer = [0; 1024];
...  |
84 | |     }
85 | | }
   | |_^
   |
   = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
  --> examples/embassy-http-server/src/main.rs:46:18
   |
46 | type AppRouter = impl picoserve::routing::PathRouter<AppState>;
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `AppRouter::{opaque#0}`, but has it in its signature

```
...
</details>